### ISSUE TITLE
Confusing error message

### ISSUE THREAD

**Issue by jpiabrantes** [huggingface/transformers#34658]
https://github.com/huggingface/transformers/blob/a06a0d12636756352494b99b5b264ac9955bc735/src/transformers/generation/utils.py#L2022

The error message says:
> A decoder-only architecture is being used, but right-padding was detected! For correct generation results, please set `padding_side='left'` when initializing the tokenizer."

But the code does not check if the input `tokenizer.padding_side == "left"`. Instead the code checks if the last token id is a padding token which is often the case when people set `tokenizer.pad_token_id = tokenizer.eos_token_id`.

**Comment by LysandreJik** [comment]
Indeed! cc @gante, @zucchini-nlp

**Comment by Rocketknight1** [comment]
gentle ping @gante @zucchini-nlp

**Comment by zucchini-nlp** [comment]
@jpiabrantes sorry for later reply. Do you mean there are cases when you want to generate from an input that ends with an `eos` token? 

The warning is mostly an advice for beginners who try to generate, since generating with right padding might result in gibberish or lower quality text.  So we point out to those who don't have much experience with generation, what are the best practices. If you already have the padding set on the correct side, you can ignore the warning

**Comment by zucchini-nlp** [comment]
@jpiabrantes yep, we could check the tokenizer's attribute directly but since tokenizer is not a compulsory kwargs when calling `generate()`, we opted to check the inputs. What I cane think now is to change the warning level in `warning.warn` to be suppressible 

>  some generations will reach eos faster than

True, but we don't try to generate from an EOS token right? When we just pass inputs to the `generate()` method, tokenizer doesn't add any EOS so it can be continued by an LLM
